scheduler/src/cook/mesos/constraints.clj

```diff
@@ -160,10 +157,11 @@
       (let [vm-resources (.getCurrAvailableResources target-vm)
             vm-attributes (get-vm-lease-attr-map vm-resources)
             [passes? reason] (job-constraint-evaluate constraint vm-resources vm-attributes
-                                                      (into (vec (.getRunningTasks target-vm))
-                                                              (mapv (fn [^com.netflix.fenzo.TaskAssignmentResult result]
-                                                                      (.getRequest result))
-                                                                     (.getTasksCurrentlyAssigned target-vm))))]
+                                                      ; Use transients to improve speed and reduce allocations
```

Comment:
Also, consider moving this expression into the let and giving the variable a name